zng/port.go

```diff
@@ -11,33 +10,15 @@ import (
 type TypeOfPort struct{}
 
 func NewPort(p uint32) Value {
-	return Value{TypePort, EncodePort(p)}
-}
-
-func EncodePort(p uint32) zcode.Bytes {
-	var b [2]byte
-	b[0] = byte(p >> 8)
-	b[1] = byte(p)
-	return b[:]
-}
-
-func DecodePort(zv zcode.Bytes) (uint32, error) {
-	if zv == nil {
-		return 0, ErrUnset
-	}
-	if len(zv) != 2 {
-		return 0, errors.New("port encoding must be 2 bytes")
-
-	}
-	return uint32(zv[0])<<8 | uint32(zv[1]), nil
+	return Value{TypePort, EncodeUint(uint64(p))}
 }
 
 func (t *TypeOfPort) Parse(in []byte) (zcode.Bytes, error) {
-	i, err := byteconv.ParseUint32(in)
+	i, err := byteconv.ParseUint64(in)
```

Comment:
Do we not need to check for i <= math.MaxUint16?

Reply:
That discussion goes in #1408.  My preference for some of this stuff (like this one) is to just let the overflow happen.